main.c

```diff
@@ -528,40 +534,64 @@ CloseApplication(HWND hwnd)
 }
 
 void
-ImportConfigFile()
+ShowConfigFileDialog(PTCHAR config_file)
 {
-    
-    TCHAR filter[2*_countof(o.ext_string)+5];
+    PTCHAR fileName;
+    int dlgresult = 0;
+
+    fileName = PathFindFileName(config_file);
+    dlgresult = ShowLocalizedMsgEx(MB_YESNO, _T(PACKAGE_NAME), IDS_ASK_IMPORT_OR_EDIT, fileName);
+    if (dlgresult == IDYES)
+        ImportConfigFile(config_file);
+    else
+        EditConfig(0, config_file);
+}
 
-    _sntprintf_0(filter, _T("*.%s%c*.%s%c"), o.ext_string, _T('\0'), o.ext_string, _T('\0'));
-    
-    OPENFILENAME fn;
+
+void
+ImportConfigFile(PTCHAR config_file)
+{
     TCHAR source[MAX_PATH] = _T("");
+    PTCHAR fileName;
+    BOOL result = 1;
 
-    fn.lStructSize = sizeof(OPENFILENAME);
-    fn.hwndOwner = NULL;
-    fn.lpstrFilter = filter;
-    fn.lpstrCustomFilter = NULL;
-    fn.nFilterIndex = 1;
-    fn.lpstrFile = source;
-    fn.nMaxFile = MAX_PATH;
-    fn.lpstrFileTitle = NULL;
-    fn.lpstrInitialDir = NULL;
-    fn.lpstrTitle = NULL;
-    fn.Flags = OFN_DONTADDTORECENT | OFN_FILEMUSTEXIST;
-    fn.lpstrDefExt = NULL;
-
-    if (GetOpenFileName(&fn)) 
+    if (config_file == NULL)
+    {
+        OPENFILENAME fn;
+        TCHAR filter[2*_countof(o.ext_string)+5];
+
+        _sntprintf_0(filter, _T("*.%s%c*.%s%c"), o.ext_string, _T('\0'), o.ext_string, _T('\0'));
+
+        fn.lStructSize = sizeof(OPENFILENAME);
+        fn.hwndOwner = NULL;
+        fn.lpstrFilter = filter;
+        fn.lpstrCustomFilter = NULL;
+        fn.nFilterIndex = 1;
+        fn.lpstrFile = source;
+        fn.nMaxFile = MAX_PATH;
+        fn.lpstrFileTitle = NULL;
+        fn.lpstrInitialDir = NULL;
+        fn.lpstrTitle = NULL;
+        fn.Flags = OFN_DONTADDTORECENT | OFN_FILEMUSTEXIST;
+        fn.lpstrDefExt = NULL;
+
+        result = GetOpenFileName(&fn) && (fileName = source + fn.nFileOffset);
+    }
+    else
     {
+        fileName = PathFindFileName(config_file);
+        _tcscpy(source, config_file);
+    }
 
+    if (result)
+    {
         TCHAR destination[MAX_PATH];
-        PTCHAR fileName = source + fn.nFileOffset;
 
         _sntprintf_0(destination, _T("%s\\%s"), o.config_dir, fileName);
 
         destination[_tcslen(destination) - _tcslen(o.ext_string) - 1] = _T('\0');
         
-        if (EnsureDirExists(destination))
+        if (EnsureDirExists(destination) && EnsureDirExists(o.config_dir))
```

Comment:
This change is not needed. If  EnsureDirExists(destination) succeeds, the next call will too as destination is a subdirectory of o.config_dir.  All necessary parent directories are automatically created by the first call. So the original code is fine.